ztools/ztls/handshake_client.go

```diff
@@ -54,6 +54,7 @@ func (c *Conn) clientHandshake() error {
 		supportedPoints:      []uint8{pointFormatUncompressed},
 		nextProtoNeg:         len(c.config.NextProtos) > 0,
 		secureRenegotiation:  true,
+		alpnProtocols:        c.config.NextProtos,
```

Comment:
I believe this was done for backward compatibility.
Before ALPN(e.g. NPN): Client says they support NPN, server responds with list of protocols, client chooses the first one from c.config.NextProtos
With ALPN: Client says they support NPN and sends c.config.NextProtos to server
So, interface hasn't changed for user